bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/WebSocketArt.java

```diff
@@ -24,140 +60,882 @@
+                    }
+                    logger.trace("{}: slideshow: {}, {}, {}, {}", host, data.getEvent(), data.getType(),
+                            data.getValue(), data.getContentId());
+                    break;
+                case "image_added":
+                    if (!data.getCategoryId().isBlank()) {
+                        logger.debug("{}: Image added: {}, category: {}", host, data.getContentId(),
+                                data.getCategoryId());
+                    }
+                    break;
+                case "get_current_artwork":
+                case "select_image":
+                case "current_artwork":
+                case "image_selected":
+                    // data.content_id: Current art displayed eg "MY_F0005"
+                    // data.is_shown: "Yes" or "No"
+                    if ("Yes".equals(data.getIsShown())) {
+                        if (!slideshow) {
+                            remoteControllerWebSocket.callback.currentAppUpdated("artMode");
+                        }
+                    }
+                    valueReceived(ART_LABEL, new StringType(data.getContentId()));
+                    if (remoteControllerWebSocket.callback.handler.isChannelLinked(ART_IMAGE)) {
+                        if (data.getEvent().contains("current_artwork") || "Yes".equals(data.getIsShown())) {
+                            getThumbnail(data.getContentId());
+                        }
+                    }
+                    break;
+                case "get_thumbnail_list":
+                case "thumbnail":
+                    logger.trace("{}: thumbnail: Fetching {}", host, data.getContentId());
+                case "ready_to_use":
+                    // upload image (should be 3840x2160 pixels in size)
+                    msg = data.getConnInfo();
+                    if (!msg.isBlank()) {
+                        JSONMessage.Contentinfo contentInfo = remoteControllerWebSocket.gson.fromJson(msg,
+                                JSONMessage.Contentinfo.class);
+                        if (contentInfo != null) {
+                            // NOTE: do not tie up the websocket receive loop for too long, so use the scheduler
+                            // upload image, or download thumbnail
+                            scheduleSocketOperation(contentInfo, "ready_to_use".equals(data.getEvent()));
+                        }
                     } else {
-                        remoteControllerWebSocket.callback.powerUpdated(true, false);
+                        // <2019 (ish) Frame TV's return thumbnails as binary data
+                        receiveThumbnail(jsonMsg.getBinaryData());
                     }
-                    remoteControllerWebSocket.updateCurrentApp();
                     break;
                 case "go_to_standby":
-                    logger.debug("go_to_standby");
+                    logger.debug("{}: go_to_standby", host);
```

Comment:
There is much similar logging in this switch statement. If it can be changed to a bit more generic logger, all these lines can be removed after placing this just above the switch start statement. It should still give enough debug information. 
```
logger.debug("{}: {}", host, data.getEvent());
```

Reply:
If I did that, then _every_ message would produce a log, and there are many messages I don't log. Bear in mind all this is reversed engineered from no documentation, and Samsung randomly changes it, So I prefer this.

Reply:
Totally understand that the logging is vital for reverse engineered communcation. I also think the binding is verbose and code can be improved. As this is not a vital part, let's leave it as is.